jetty-ee9/jetty-ee9-webapp/src/test/java/org/eclipse/jetty/ee9/webapp/WebAppContextTest.java

```diff
@@ -546,6 +551,132 @@ public void testBaseResourceAbsolutePath() throws Exception
         assertTrue(context.isAvailable(), "WebAppContext should be available");
     }
 
+    @Test
+    public void testGetResourceFromCollection() throws Exception
+    {
+        Server server = newServer();
+
+        WebAppContext context = new WebAppContext();
+        context.setContextPath("/");
+        context.setBaseResource(ResourceFactory.combine(
+            ResourceFactory.root().newResource(MavenPaths.findTestResourceDir("wars/layer0/")),
+            ResourceFactory.root().newResource(MavenPaths.findTestResourceDir("wars/layer1/"))));
+        server.setHandler(context);
+        server.start();
+
+        ServletContext servletContext = context.getServletContext();
+        assertThat(servletContext.getResource("/WEB-INF/zero.xml"), notNullValue());
+        assertThat(servletContext.getResource("/WEB-INF/one.xml"), notNullValue());
+    }
+
+    @Test
+    public void testGetResourcePathsFromCollection() throws Exception
+    {
+        Server server = newServer();
+
+        WebAppContext context = new WebAppContext();
+        context.setContextPath("/");
+        context.setBaseResource(ResourceFactory.combine(
+            ResourceFactory.root().newResource(MavenPaths.findTestResourceDir("wars/layer0/")),
+            ResourceFactory.root().newResource(MavenPaths.findTestResourceDir("wars/layer1/"))));
+        server.setHandler(context);
+        server.start();
+
+        ServletContext servletContext = context.getServletContext();
+        assertThat(servletContext.getResourcePaths("/WEB-INF/"), containsInAnyOrder("/WEB-INF/zero.xml", "/WEB-INF/one.xml"));
+    }
+
+    @Test
+    public void testGetResourcePathsWithDirsFromCollection() throws Exception
+    {
+        Server server = newServer();
+
+        WebAppContext context = new WebAppContext();
+        context.setContextPath("/");
+        context.setBaseResource(ResourceFactory.combine(
+            ResourceFactory.root().newResource(MavenPaths.findTestResourceDir("wars/layer0/")),
+            ResourceFactory.root().newResource(MavenPaths.findTestResourceDir("wars/layer1/")),
+            ResourceFactory.root().newResource(MavenPaths.findTestResourceDir("wars/with_dirs/"))
```

Comment:
Please avoid using `ResourceFactory.root()` when `context.getResourceFactory()` is as easy as well as better practice.